pkgs/by-name/gh/ghostty/package.nix

```
@@ -1,199 +1,37 @@
-
-      "-Dapp-runtime=${appRuntime}"
-      "-Dfont-backend=${fontBackend}"
-      "-Dgtk-adwaita=${lib.boolToString withAdwaita}"
-      "-Drenderer=${renderer}"
-    ]
-    ++ lib.mapAttrsToList (name: package: "-fsys=${name} --search-prefix ${lib.getLib package}") {
-      inherit glslang;
-    };
-
-  zigCheckFlags = finalAttrs.zigBuildFlags;
-
-  doCheck = true;
-
-  /**
-    Ghostty really likes all of it's resources to be in the same directory, so link them back after we split them
-
-    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/os/resourcesdir.zig#L11-L52
-    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L745-L750
-    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L818-L834
-
-    terminfo and shell integration should also be installable on remote machines
-
-    ```nix
-    { pkgs, ... }: {
-      environment.systemPackages = [ pkgs.ghostty.terminfo ];
-
-      programs.bash = {
-        interactiveShellInit = ''
-          if [[ "$TERM" == "xterm-ghostty" ]]; then
-            builtin source ${pkgs.ghostty.shell_integration}/bash/ghostty.bash
-          fi
-        '';
-      };
-    }
-    ```
-  */
-  postFixup = ''
-    ln -s $man/share/man $out/share/man
-
-    moveToOutput share/terminfo $terminfo
-    ln -s $terminfo/share/terminfo $out/share/terminfo
-
-    mv $out/share/ghostty/shell-integration $shell_integration
-    ln -s $shell_integration $out/share/ghostty/shell-integration
-
-    mv $out/share/vim/vimfiles $vim
-    rmdir $out/share/vim
-    ln -s $vim $out/share/vim-plugins
-
-
-    remove-references-to -t ${finalAttrs.deps} $out/bin/.ghostty-wrapped
-  '';
-
-  nativeInstallCheckInputs = [
-    versionCheckHook
-  ];
-
-  doInstallCheck = true;
-
-  versionCheckProgramArg = [ "--version" ];
-
-  passthru = {
-    tests = lib.optionalAttrs stdenv.hostPlatform.isLinux {
-      inherit (nixosTests) allTerminfo;
-      nixos = nixosTests.terminal-emulators.ghostty;
+if settings == { } then
+  ghostty-unwrapped
+else
+  writeShellApplication {
+    name = "ghostty";
+
+    derivationArgs = {
+      inherit (ghostty-unwrapped) pname version meta;
     };
-  };
 
-  meta = {
-    description = "Fast, native, feature-rich terminal emulator pushing modern features";
-    longDescription = ''
-      Ghostty is a terminal emulator that differentiates itself by being
-      fast, feature-rich, and native. While there are many excellent terminal
-      emulators available, they all force you to choose between speed,
-      features, or native UIs. Ghostty provides all three.
+    text = ''
+      options=( ${lib.concatStringsSep " " (lib.mapAttrsToList mkFlag settings)} )
+      if [[ $# -gt 0 && "$1" == "+"* ]]; then
+        ${lib.getExe ghostty-unwrapped} "$@"
+      else
+        ${lib.getExe ghostty-unwrapped} "''${options[@]}" "$@"
+      fi
     '';
```

```suggestion
  postBuild =
    ''
      wrapProgram $out/bin/ghostty \
        "''${gappsWrapperArgs[@]}" \
        ${lib.optionalString (settings != {}) "--add-flags '${lib.escapeShellArgs configFlags}'"}
      versionCheckHook
    ''
    # Link back to the unwrapped package's split outputs
    + lib.concatLines (
      map (output: "ln -s ${ghostty-unwrapped.${output}} \$${output}") (
        lib.remove "out" ghostty-unwrapped.outputs
      )
    );
    
  passthru = {
    inherit ghostty-unwrapped;

    tests = lib.optionalAttrs stdenv.hostPlatform.isLinux {
      inherit (nixosTests) allTerminfo;
      nixos = nixosTests.terminal-emulators.ghostty;
    };
  } // lib.optionalAttrs (settings != { }) { inherit configFlags; };
```
Let's also move the testing over here to make sure the wrapper works